iconic/UserProfile/views.py

```diff
@@ -19,6 +19,47 @@ def get(self, request):
         return Response(data, status=status.HTTP_200_OK)
 
 
+# A view for reading the number of followers
+class NumberOfFollowersView(APIView):
+    def get(self, request):
+        number_of_followers = Followers.objects.filter(user_id=request.user.id).count()
+        return Response({"user_id": request.user.id, "number_of_followers": number_of_followers},
+                        status=status.HTTP_200_OK)
+
+
+# A view for getting, adding and deleting followers
+class FollowersGetAddDeleteView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):
+        user_id = request.data['user_id']
+        follower_id = request.user.id
+        current_follower = Followers.objects.get(user_id=user_id, follower=follower_id)
+        if current_follower is not None:
+            data = CustomFollowersSerializer(current_follower).data
+            return Response(data, status=status.HTTP_200_OK)
+        else:
+            return Response({"message": "Пользователь не подписан на данного пользователя"},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    #  for adding
+    def post(self, request):
+        user_id = request.data['user_id']
+        follower_id = request.user.id
+        current_follower = Followers(user_id=user_id, follower=follower_id)
+        data = CustomFollowersSerializer(current_follower).data
+        serializer = CustomFollowersSerializer(data=data)
```

Comment:
Do not create Followers instance, just create data dict and put "user_id" and "follower" in, than put this data in serializer